packages/react/src/components/Tag/index.tsx

```diff
@@ -18,14 +45,79 @@ TagLabel.propTypes = {
   className: PropTypes.string
 };
 
-const Tag = ({ children, className, ...other }: TagProps) => (
-  <div className={classNames('Tag', className)} {...other}>
-    {children}
-  </div>
-);
+const Tag = (props: TagProps) => {
+  // render dismiss Tag
+  if (props.variant === 'dismiss') {
+    const [show, setShow] = useState(true);
+    const { children, className, onDismiss, ...other } = props;
+    const handleDismiss = (event: React.MouseEvent<HTMLButtonElement>) => {
+      setShow(false);
+      if (onDismiss) {
+        onDismiss(event);
+      }
+    };
+
+    return (
+      <button
+        className={classNames('Tag', 'Tag--dismiss', className, {
+          'Tag--hidden': !show
+        })}
+        onClick={handleDismiss}
+        {...other}
+      >
+        {children}
+        <Icon type="close" />
+      </button>
+    );
+  }
+
+  // render toggle Tag
+  if (props.variant === 'toggle') {
+    const {
+      children,
+      className,
+      toggleState,
+      buttonLabel,
+      onToggle,
+      toggleOnText = 'ON',
+      toggleOffText = 'OFF',
```

Comment:
Two things:

1. Will there always be toggle text? If so, I suppose this is okay. If not, though, we may want to leave it up to the children that are passed in.
2. It doesn't look like we have i18n in cauldron, so I am concerned about how this will get translated.

Reply:
> It doesn't look like we have i18n in cauldron, so I am concerned about how this will get translated.

Isn't that precisely why we'd want to support a prop (or as you said children)?

Reply:
@schne324 What I meant by that comment was if we did have specific toggle props, then we should not provide a default value for them of `'ON'` and `'OFF'` because they would not be translated. If we do have separate props, then they should not have a default value because. However, if we just rely on `children` alone, that point is moot.

Reply:
To follow up a bit on this, @scurker pointed out that we do have default strings elsewhere (see: https://github.com/dequelabs/cauldron/pull/501#discussion_r783125464).

However, the point still stands here that I think this should just come in via children anyway, rather than standalone props.

Reply:
>  I think this should just come in via children

sounds good to me